docs/guide/api-environment-runtimes.md

```diff
@@ -363,6 +381,8 @@ await createServer({
 
 :::
 
+Make sure to implement the `connection` / `vite:client:connect` / `vite:client:disconnect` events in the `on` / `off` methods when those methods exist. `connection` event and `vite:client:connect` event should be emitted when the connection is established, and `vite:client:disconnect` event should be emitted when the connection is closed. The `HotChannelClient` object passed to the event handler must have the same reference for the same connection.
```

Comment:
When reading the updated example, I don't understand why there's the `connection` event alongside `vite:client:connect`. Can we explain when `connection` is triggered?


From what I understand, `connection` feels like "ready state" and it can start accepting client connections? I guess it's a bit late to rename it now as it's shared/used by the websocket server setup as the transport. `vite:ready` would be nice.